crates/typst/src/symbols/symbol.rs

```diff
@@ -215,6 +204,41 @@ impl Display for Symbol {
     }
 }
 
+impl SymChar {
+    /// Create a symbol character without a function.
+    pub const fn pure(c: char) -> Self {
+        Self::Pure(c)
+    }
+
+    /// Create a symbol character with a function.
+    pub const fn with_func(c: char, func: fn() -> Func) -> Self {
+        Self::WithFunc(c, func)
+    }
+
+    /// Get the function associated with the symbol.
+    pub fn func(&self) -> Option<Func> {
+        match self {
+            Self::Pure(_) => None,
+            Self::WithFunc(_, func) => Some((func)()),
+        }
+    }
+}
+
+impl From<&SymChar> for char {
```

Comment:
I think an accessor would be clearer for this. The `char -> SymChar` from impl is fine, but this one feels a bit off.